ffi/wgpu.h

```diff
@@ -298,6 +295,26 @@ typedef enum {
   WGPUVertexFormat_Int4 = 29,
 } WGPUVertexFormat;
 
+/**
+ * This type is not to be constructed by any users of wgpu. If you construct this type, any semver
+ * guarantees made by wgpu are invalidated and a non-breaking change may break your code.
+ *
+ * If you are here trying to construct it, the solution is to use partial construction with the
+ * default:
+ *
+ * ```ignore
+ * let limits = Limits {
+ *     max_bind_groups: 2,
+ *     ..Limits::default()
+ * }
+ * ```
+ */
+typedef struct WGPUNonExhaustive WGPUNonExhaustive;
+
+typedef struct WGPUSamplerDescriptor WGPUSamplerDescriptor;
+
+typedef struct WGPUUnsafeExtensions WGPUUnsafeExtensions;
```

Comment:
This needs a C compatible analogue made, possibly manifesting as a raw boolean instead of this opaque type.